### ISSUE TITLE
Authentication Activity Leaks 

### ISSUE THREAD

**Issue by ravarora** [AzureAD/azure-activedirectory-library-for-android#1147]
Please ensure the following information is provided when logging a new issue:

<ul>
  <li>Expected behavior</li>
        After the authentication activity (web sign in) finishes either by successful log-in or by user cancelling the operation. the Authentication Activity instance is not getting cleared from the memory. 

  <li>Actual behavior</li>
        After the activity finishes its instance should be removed from the memory. 
  <li>Steps to reproduce the problem</li>
        Used android profiler to analyse memory dump before starting and after closing the Authentication Activity. 
  <li>Specifications like the version of the library, operating system etc</li>
         Adal version 1.1.13
  <li>Relevant logs and traces</li>

In com.microsoft.mobile.polymer.dev1:1.0.9219:1692190001.
* com.microsoft.aad.adal.AuthenticationActivity has leaked:
* GC ROOT thread java.lang.Thread.<Java Local> (named 'Thread-6')
* references com.android.tools.profiler.support.event.InputConnectionWrapper.mTarget
* references org.chromium.content.browser.input.ThreadedInputConnection.mTargetView
* references android.webkit.WebView.mContext
* leaks com.microsoft.aad.adal.AuthenticationActivity instance

* Retaining: 64 kB.
* Reference Key: 2c877109-df3c-4a24-873f-40ebcc4a9f94
* Device: Xiaomi xiaomi Mi A1 tissot
* Android Version: 8.0.0 API: 26 LeakCanary: 1.5.4 74837f0
* Durations: watch=6080ms, gc=375ms, heap dump=2450ms, analysis=91902ms

* Details:
* Instance of java.lang.Thread
|   static uncaughtExceptionPreHandler = com.android.internal.os.RuntimeInit$LoggingHandler@315942664 (0x12d4e708)
|   static SUBCLASS_IMPLEMENTATION_PERMISSION = java.lang.RuntimePermission@1884603248 (0x7054c370)
|   static threadInitNumber = 51
|   static EMPTY_STACK_TRACE = java.lang.StackTraceElement[0]@1884603232 (0x7054c360)
|   static MIN_PRIORITY = 1
|   static $classOverhead = byte[232]@1884605073 (0x7054ca91)
|   static MAX_PRIORITY = 10
|   static NANOS_PER_MILLI = 1000000
|   static NORM_PRIORITY = 5
|   static threadSeqNumber = 593
|   static defaultUncaughtExceptionHandler = com.android.internal.os.RuntimeInit$KillApplicationHandler@315942656 (0x12d4e700)
|   blocker = null
|   blockerLock = java.lang.Object@320771872 (0x131e9720)
|   contextClassLoader = dalvik.system.PathClassLoader@316057488 (0x12d6a790)
|   daemon = false
|   eetop = 0
|   group = java.lang.ThreadGroup@1884255056 (0x704f7350)
|   inheritableThreadLocals = null
|   inheritedAccessControlContext = java.security.AccessControlContext@320771880 (0x131e9728)
|   lock = java.lang.Object@315890464 (0x12d41b20)
|   name = java.lang.String@320771888 (0x131e9730)
|   nativeParkEventPointer = 0
|   nativePeer = 3480763392
|   parkBlocker = null
|   parkState = 1
|   priority = 5
|   single_step = false
|   stackSize = 0
|   started = true
|   stillborn = false
|   target = com.android.tools.profiler.support.profilers.EventProfiler$InputConnectionHandler@315890848 (0x12d41ca0)
|   threadLocalRandomProbe = 0
|   threadLocalRandomSecondarySeed = 0
|   threadLocalRandomSeed = 0
|   threadLocals = java.lang.ThreadLocal$ThreadLocalMap@320771912 (0x131e9748)
|   threadQ = null
|   threadStatus = 0
|   tid = 192
|   uncaughtExceptionHandler = null
|   shadow$_klass_ = java.lang.Thread
|   shadow$_monitor_ = 0
* Instance of com.android.tools.profiler.support.event.InputConnectionWrapper
|   static $classOverhead = byte[164]@316313273 (0x12da8eb9)
|   mMissingMethodFlags = 0
|   mMutable = false
|   mTarget = org.chromium.content.browser.input.ThreadedInputConnection@320656872 (0x131cd5e8)
|   shadow$_klass_ = com.android.tools.profiler.support.event.InputConnectionWrapper
|   shadow$_monitor_ = 0
* Instance of org.chromium.content.browser.input.ThreadedInputConnection
|   static UNBLOCKER = org.chromium.content.browser.input.ThreadedInputConnection$1@318437416 (0x12faf828)
|   static $classOverhead = byte[192]@317444921 (0x12ebd339)
|   mCachedTextInputState = org.chromium.content.browser.input.TextInputState@320657584 (0x131cd8b0)
|   mCurrentExtractedTextRequestToken = 0
|   mFinishComposingTextRunnable = org.chromium.content.browser.input.ThreadedInputConnection$5@320657608 (0x131cd8c8)
|   mHandler = android.os.Handler@318437560 (0x12faf8b8)
|   mImeAdapter = org.chromium.content.browser.input.ImeAdapter@320657624 (0x131cd8d8)
|   mNotifyUserActionRunnable = org.chromium.content.browser.input.ThreadedInputConnection$4@320657720 (0x131cd938)
|   mNumNestedBatchEdits = 0
|   mPendingAccent = 0
|   mProcessPendingInputStatesRunnable = org.chromium.content.browser.input.ThreadedInputConnection$2@320657736 (0x131cd948)
|   mQueue = java.util.concurrent.LinkedBlockingQueue@320657752 (0x131cd958)
|   mRequestTextInputStateUpdate = org.chromium.content.browser.input.ThreadedInputConnection$3@320657792 (0x131cd980)
|   mShouldUpdateExtractedText = false
|   mDefaultComposingSpans = null
|   mDummyMode = false
|   mEditable = null
|   mIMM = android.view.inputmethod.InputMethodManager@315890656 (0x12d41be0)
|   mKeyCharacterMap = null
|   mTargetView = android.webkit.WebView@320656952 (0x131cd638)
|   shadow$_klass_ = org.chromium.content.browser.input.ThreadedInputConnection
|   shadow$_monitor_ = 0
* Instance of android.webkit.WebView
|   static SCHEME_MAILTO = java.lang.String@1889331752 (0x709cea28)
|   static RENDERER_PRIORITY_BOUND = 1
|   static RENDERER_PRIORITY_WAIVED = 0
|   static DATA_REDUCTION_PROXY_SETTING_CHANGED = java.lang.String@315982064 (0x12d580f0)
|   static SCHEME_TEL = java.lang.String@1887743416 (0x7084adb8)
|   static RENDERER_PRIORITY_IMPORTANT = 2
|   static $classOverhead = byte[4193]@316035409 (0x12d65151)
|   static LOGTAG = java.lang.String@1889611896 (0x70a13078)
|   static sEnforceThreadChecking = true
|   static SCHEME_GEO = java.lang.String@1889666768 (0x70a206d0)
|   mFindListener = null
|   mProvider = com.android.webview.chromium.WebViewChromium@320767968 (0x131e87e0)
|   mWebViewThread = android.os.Looper@315883616 (0x12d40060)
|   mAnimationListener = null
|   mCachePaint = null
|   mChildCountWithTransientState = 0
|   mChildTransformation = null
|   mChildren = android.view.View[12]@320768960 (0x131e8bc0)
|   mChildrenCount = 0
|   mChildrenInterestedInDrag = null
|   mCurrentDragChild = null
|   mCurrentDragStartEvent = null
|   mDefaultFocus = null
|   mDisappearingChildren = null
|   mFirstHoverTarget = null
|   mFirstTouchTarget = null
|   mFocused = null
|   mFocusedInCluster = null
|   mGroupFlags = 2244691
|   mHoveredSelf = false
|   mInvalidateRegion = null
|   mInvalidationTransformation = null
|   mIsInterestedInDrag = false
|   mLastTouchDownIndex = -1
|   mLastTouchDownTime = 0
|   mLastTouchDownX = 0.0
|   mLastTouchDownY = 0.0
|   mLayoutAnimationController = null
|   mLayoutCalledWhileSuppressed = false
|   mLayoutMode = -1
|   mLayoutTransitionListener = android.view.ViewGroup$2@320769024 (0x131e8c00)
|   mLocalPoint = null
|   mNestedScrollAxes = 0
|   mOnHierarchyChangeListener = null
|   mPersistentDrawingCache = 2
|   mPreSortedChildren = null
|   mSuppressLayout = false
|   mTempPoint = null
|   mTooltipHoverTarget = null
|   mTooltipHoveredSelf = false
|   mTransientIndices = null
|   mTransientViews = null
|   mTransition = null
|   mTransitioningViews = null
|   mVisibilityChangingChildren = null
|   mAccessibilityCursorPosition = -1
|   mAccessibilityDelegate = null
|   mAccessibilityTraversalAfterId = -1
|   mAccessibilityTraversalBeforeId = -1
|   mAccessibilityViewId = -1
|   mAnimator = null
|   mAttachInfo = null
|   mAttributes = null
|   mAutofillHints = null
|   mAutofillId = android.view.autofill.AutofillId@318428936 (0x12fad708)
|   mAutofillViewId = 1073741824
|   mBackground = null
|   mBackgroundRenderNode = null
|   mBackgroundResource = 0
|   mBackgroundSizeChanged = true
|   mBackgroundTint = null
|   mBottom = 1848
|   mCachingFailed = false
|   mClipBounds = null
|   mContentDescription = null
|   mContext = com.microsoft.aad.adal.AuthenticationActivity@320658424 (0x131cdbf8)
|   mCurrentAnimation = null
|   mDefaultFocusHighlight = null
|   mDefaultFocusHighlightCache = null
|   mDefaultFocusHighlightEnabled = true
|   mDefaultFocusHighlightSizeChanged = true
|   mDrawableState = int[3]@1888134744 (0x708aa658)
|   mDrawingCache = null
|   mDrawingCacheBackgroundColor = 0
|   mFloatingTreeObserver = null
|   mForegroundInfo = null
|   mFrameMetricsObservers = null
|   mGhostView = null
|   mHasPerformedLongPress = false
|   mID = 2131755184
|   mIgnoreNextUpEvent = false
|   mInContextButtonPress = false
|   mInputEventConsistencyVerifier = null
|   mKeyedTags = null
|   mLabelForId = -1
|   mLastIsOpaque = false
|   mLayerPaint = null
|   mLayerType = 0
|   mLayoutInsets = null
|   mLayoutParams = android.widget.RelativeLayout$LayoutParams@320769040 (0x131e8c10)
|   mLeft = 0
|   mLeftPaddingDefined = false
|   mListenerInfo = android.view.View$ListenerInfo@320769120 (0x131e8c60)
|   mLongClickX = NaN
|   mLongClickY = NaN
|   mMatchIdPredicate = null
|   mMatchLabelForPredicate = null
|   mMeasureCache = android.util.LongSparseLongArray@320769192 (0x131e8ca8)
|   mMeasuredHeight = 1848
|   mMeasuredWidth = 16778296
|   mMinHeight = 0
|   mMinWidth = 0
|   mNestedScrollingParent = null
|   mNextClusterForwardId = -1
|   mNextFocusDownId = -1
|   mNextFocusForwardId = -1
|   mNextFocusLeftId = -1
|   mNextFocusRightId = -1
|   mNextFocusUpId = -1
|   mOldHeightMeasureSpec = 1073743672
|   mOldWidthMeasureSpec = 1073742904
|   mOutlineProvider = android.view.ViewOutlineProvider$1@1887940568 (0x7087afd8)
|   mOverScrollMode = 1
|   mOverlay = null
|   mPaddingBottom = 0
|   mPaddingLeft = 0
|   mPaddingRight = 0
|   mPaddingTop = 0
|   mParent = android.widget.RelativeLayout@320682056 (0x131d3848)
|   mPendingCheckForLongPress = null
|   mPendingCheckForTap = null
|   mPerformClick = null
|   mPointerIcon = null
|   mPrivateFlags = -2128082926
|   mPrivateFlags2 = 1610819112
|   mPrivateFlags3 = 524288
|   mRecreateDisplayList = false
|   mRenderNode = android.view.RenderNode@320769216 (0x131e8cc0)
|   mResources = android.content.res.Resources@320660848 (0x131ce570)
|   mRight = 1080
|   mRightPaddingDefined = false
|   mRoundScrollbarRenderer = null
|   mRunQueue = android.view.HandlerActionQueue@320769240 (0x131e8cd8)
|   mScrollCache = android.view.View$ScrollabilityCache@320769256 (0x131e8ce8)
|   mScrollIndicatorDrawable = null
|   mScrollX = 0
|   mScrollY = 0
|   mSendViewScrolledAccessibilityEvent = null
|   mSendViewStateChangedAccessibilityEvent = null
|   mSendingHoverAccessibilityEvents = false
|   mStartActivityRequestWho = null
|   mStateListAnimator = null
|   mSystemUiVisibility = 0
|   mTag = null
|   mTempNestedScrollConsumed = null
|   mTooltipInfo = null
|   mTop = 0
|   mTouchDelegate = null
|   mTouchSlop = 24
|   mTransformationInfo = null
|   mTransientStateCount = 0
|   mTransitionName = null
|   mUnscaledDrawingCache = null
|   mUnsetPressedState = null
|   mUserPaddingBottom = 0
|   mUserPaddingEnd = -2147483648
|   mUserPaddingLeft = 0
|   mUserPaddingLeftInitial = 0
|   mUserPaddingRight = 0
|   mUserPaddingRightInitial = 0
|   mUserPaddingStart = -2147483648
|   mVerticalScrollFactor = 0.0
|   mVerticalScrollbarPosition = 0
|   mViewFlags = 402932481
|   mVisibilityChangeForAutofillHandler = null
|   mWindowAttachCount = 1
|   shadow$_klass_ = android.webkit.WebView
|   shadow$_monitor_ = 0
* Instance of com.microsoft.aad.adal.AuthenticationActivity
|   static TAG = java.lang.String@316266656 (0x12d9d8a0)
|   static $classOverhead = byte[1928]@317290969 (0x12e979d9)
|   static BACK_PRESSED_CANCEL_DIALOG_STEPS = -2
|   mAccountAuthenticatorResponse = null
|   mAuthRequest = com.microsoft.aad.adal.AuthenticationRequest@320658712 (0x131cdd18)
|   mAuthenticatorResultBundle = null
|   mCallingPackage = null
|   mCallingUID = 0
|   mJWSBuilder = com.microsoft.aad.adal.JWSBuilder@320658792 (0x131cdd68)
|   mPkeyAuthRedirect = false
|   mReceiver = com.microsoft.aad.adal.AuthenticationActivity$ActivityBroadcastReceiver@320658800 (0x131cdd70)
|   mRedirectUrl = java.lang.String@320658824 (0x131cdd88)
|   mRegisterReceiver = true
|   mSpinner = android.app.ProgressDialog@320658920 (0x131cdde8)
|   mStartUrl = java.lang.String@320659072 (0x131cde80)
|   mStorageHelper = com.microsoft.aad.adal.StorageHelper@320659608 (0x131ce098)
|   mUIEvent = com.microsoft.aad.adal.UIEvent@320659648 (0x131ce0c0)
|   mWaitingRequestId = 0
|   mWebRequestHandler = com.microsoft.aad.adal.WebRequestHandler@320659672 (0x131ce0d8)
|   mWebView = android.webkit.WebView@320656952 (0x131cd638)
|   mActionBar = null
|   mActionModeTypeStarting = 0
|   mActivityInfo = android.content.pm.ActivityInfo@320659688 (0x131ce0e8)
|   mActivityTransitionState = android.app.ActivityTransitionState@320659840 (0x131ce180)
|   mApplication = com.microsoft.mobile.polymer.PolymerApplication@316057320 (0x12d6a6e8)
|   mAutoFillResetNeeded = true
|   mAutofillManager = android.view.autofill.AutofillManager@320659896 (0x131ce1b8)
|   mAutofillPopupWindow = null
|   mCalled = true
|   mChangeCanvasToTranslucent = false
|   mChangingConfigurations = false
|   mComponent = android.content.ComponentName@318428360 (0x12fad4c8)
|   mConfigChangeFlags = 0
|   mCurrentConfig = android.content.res.Configuration@320659952 (0x131ce1f0)
|   mDecor = null
|   mDefaultKeyMode = 0
|   mDefaultKeySsb = null
|   mDestroyed = true
|   mDoReportFullyDrawn = false
|   mEmbeddedID = null
|   mEnableDefaultActionBarUp = false
|   mEnterTransitionListener = android.app.SharedElementCallback$1@1887914224 (0x708748f0)
|   mExitTransitionListener = android.app.SharedElementCallback$1@1887914224 (0x708748f0)
|   mFinished = true
|   mFragments = android.app.FragmentController@320660064 (0x131ce260)
|   mHandler = android.os.Handler@320660080 (0x131ce270)
|   mHasCurrentPermissionsRequest = false
|   mIdent = 18307273
|   mInstanceTracker = android.os.StrictMode$InstanceTracker@320660112 (0x131ce290)
|   mInstrumentation = android.app.Instrumentation@317521560 (0x12ecfe98)
|   mIntent = android.content.Intent@320660128 (0x131ce2a0)
|   mLastAutofillId = 1073741828
|   mLastNonConfigurationInstances = null
|   mMainThread = android.app.ActivityThread@315883712 (0x12d400c0)
|   mManagedCursors = java.util.ArrayList@320660192 (0x131ce2e0)
|   mManagedDialogs = null
|   mMenuInflater = null
|   mParent = null
|   mReferrer = java.lang.String@320660216 (0x131ce2f8)
|   mResultCode = 2002
|   mResultData = android.content.Intent@320660272 (0x131ce330)
|   mResumed = false
|   mSearchEvent = null
|   mSearchManager = null
|   mStartedActivity = false
|   mStopped = true
|   mTaskDescription = android.app.ActivityManager$TaskDescription@320660336 (0x131ce370)
|   mTemporaryPause = false
|   mTitle = java.lang.String@317475600 (0x12ec4b10)
|   mTitleColor = 0
|   mTitleReady = true
|   mToken = android.os.BinderProxy@320660376 (0x131ce398)
|   mTranslucentCallback = null
|   mUiThread = java.lang.Thread@1933854224 (0x73444610)
|   mVisibleBehind = false
|   mVisibleFromClient = true
|   mVisibleFromServer = true
|   mVoiceInteractor = null
|   mWindow = com.android.internal.policy.PhoneWindow@320660408 (0x131ce3b8)
|   mWindowAdded = true
|   mWindowManager = android.view.WindowManagerImpl@320660776 (0x131ce528)
|   mInflater = com.android.internal.policy.PhoneLayoutInflater@320660800 (0x131ce540)
|   mOverrideConfiguration = null
|   mResources = android.content.res.Resources@320660848 (0x131ce570)
|   mTheme = android.content.res.Resources$Theme@320660888 (0x131ce598)
|   mThemeResource = 2131493040
|   mBase = android.app.ContextImpl@320660904 (0x131ce5a8)
|   shadow$_klass_ = com.microsoft.aad.adal.AuthenticationActivity
|   shadow$_monitor_ = 1289457794
* Excluded Refs:
| Field: android.view.textservice.SpellCheckerSession$1.this$0
| Field: android.view.Choreographer$FrameDisplayEventReceiver.mMessageQueue (always)
| Thread:FinalizerWatchdogDaemon (always)
| Thread:main (always)
| Thread:LeakCanary-Heap-Dump (always)
| Class:java.lang.ref.WeakReference (always)
| Class:java.lang.ref.SoftReference (always)
| Class:java.lang.ref.PhantomReference (always)
| Class:java.lang.ref.Finalizer (always)
| Class:java.lang.ref.FinalizerReference (always)
</ul>

**Comment by iambmelt** [comment]
Partially fixed by:
https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1511

Fix will roll-out in a future ADAL release
